redisinsight/ui/src/slices/app/csrf.ts

```diff
@@ -0,0 +1,58 @@
+import { createSlice } from '@reduxjs/toolkit'
+import apiService, { setApiCsrfHeader } from 'uiSrc/services/apiService'
+import { setResourceCsrfHeader } from 'uiSrc/services/resourcesService'
+import { AppDispatch, RootState } from '../store'
+
+const getCsrfEndpoint = () => process.env.RI_CSRF_ENDPOINT || ''
+
+interface CSRFTokenResponse {
+  token: string;
+}
+
+export const initialState: {
+  csrfEndpoint: string;
+  loading: boolean;
+  token: string;
+} = {
+  csrfEndpoint: getCsrfEndpoint(),
+  loading: true,
```

Comment:
I think we should have loading: false by default, and set it to true when we fetching.
and then we can use `(!loading || token) ? children : <PagePlaceholder />` 

but it this case need to think what should be on refetch token (just replace token without deleting?)

Reply:
I've updated the initial state so that loading = false, but the logic had to be adjusted somewhat. If the logic was just as you've written, children would be rendered on the first render of the csrf component. The csrf component is made to prevent that from happening because rendering the children will trigger additional web requests, and in a cloud environment, those web requests will fail if the csrf token is not available. So, near the app root level, we effectively block everything from rendering until we have the csrf token, then we can safely render the rest of the app.